orttraining/orttraining/python/training/__init__.py

```diff
@@ -3,24 +3,30 @@
 # Licensed under the MIT License.
 # --------------------------------------------------------------------------
 # isort: skip_file
+import importlib.util
+
 from onnxruntime.capi._pybind_state import (
     PropagateCastOpsStrategy,
     TrainingParameters,
     is_ortmodule_available,
 )
 
 # Options need to be imported before `ORTTrainer`.
-from . import amp, artifacts, optim
+from . import amp, artifacts
 
 __all__ = [
     "PropagateCastOpsStrategy",
     "TrainingParameters",
     "is_ortmodule_available",
     "amp",
     "artifacts",
-    "optim",
 ]
 
+if importlib.util.find_spec("torch") is not None:
+    from . import optim
+
+    __all__ += ["optim"]
```

Comment:
.append(...)?

Reply:
Yes, that would work as well. I'm following the example in the lines below, where `ORTModule` is added to `__all__` as `__all__ += ["ORTModule"]`.